docs/deprecation_policy.md

```diff
@@ -0,0 +1,37 @@
+Deprecation Policy for Platform Dependencies
+============================================
+
+Synapse has a number of platform dependencies, including Python and PostgreSQL.
+This document outlines the policy towards which versions we support, and how we
+drop support for versions in the future.
+
+
+Policy
+------
+
+For Python and PostgreSQL Synapse will follow the upstream support life cycles,
+i.e. when a version reaches End of Life Synapse will withdraw support for that
+version in future releases.
+
+The team will not intentionally break compatibility for at least one month after
+an EOL date, but does explicitly reserve the right to do so should security or
+other concerns arise in an unmaintained version of a dependency.
```

Comment:
...do we _actually_ want to box ourselves in to this, or should we drop when upstream drops?

Reply:
I'd me more than happy to drop this, it seems a bit pointless.

Reply:
Make it so